### PR TITLE
Created test for matrix operations

### PR DESCRIPTION
**Pull request by Sofasmile**

### CODE REVIEW

File: src/test/java/core/basesyntax/MatrixOperationTest.java [hunk 1]
```diff
@@ -1,8 +1,44 @@
...
-public class HelloWorldTest {
+import org.junit.Assert;
+import org.junit.Test;
 
-}
\ No newline at end of file
+import java.util.Arrays;
+
+public class MatrixOperationTest {
+    private static final int[][] TEST_MATRIX = {
```
Comment (boroda4436) [inline src/test/java/core/basesyntax/MatrixOperationTest.java]: @Sofasmile add at least one more example to avoid hardcoding the return result by students

Comment (boroda4436) [reply src/test/java/core/basesyntax/MatrixOperationTest.java]: @Sofasmile you may use Parametrized test, but it is not required

File: src/test/java/core/basesyntax/MatrixOperationTest.java [hunk 2]
```diff
@@ -1,8 +1,81 @@
...
+import java.util.Arrays;
+
+public class MatrixOperationTest {
+    private static final int[][] FIRST_TEST_MATRIX = {
+            {2, 31, 6, 7},
+            {15, 42, 1, 0},
+            {9, 7, 12, 19},
+            {55, 34, 1, 10}
+    };
+    private static final int[][] SECOND_TEST_MATRIX = {
```
Comment (roman-mate) [inline src/test/java/core/basesyntax/MatrixOperationTest.java]: what's the conceptual difference between first and second matrix?

File: src/test/java/core/basesyntax/MatrixOperationTest.java [hunk 3]
```diff
@@ -1,8 +1,81 @@
...
+    private static final int[][] THIRD_TEST_MATRIX = {
+            {-5}
+    };
+
+    private static final int[] FIRST_EXPECTED_DIAGONAL = {2, 42, 12, 10};
+    private static final int[] SECOND_EXPECTED_DIAGONAL = {-2, 0, -12};
+    private static final int[] THIRD_EXPECTED_DIAGONAL = {-5};
+
+    @Test
+    public void getDiagonal() {
```
Comment (roman-mate) [inline src/test/java/core/basesyntax/MatrixOperationTest.java]: You need to break down method behavior into different scenarios (if they exist). And write test case per scenario. We should not be testing different scenarios in one test case.

File: src/test/java/core/basesyntax/MatrixOperationTest.java (hunk 2 above)
Comment (Sofasmile) [reply src/test/java/core/basesyntax/MatrixOperationTest.java]: there isn't the conceptual difference
only different size of matrices

**Pull request merged**